File: solutions/other/indigo-code-2023/bblock.cpp
#include <bits/stdc++.h>
using namespace std;

#pragma GCC optimize("O3,unroll-loops")
int main() {
    int n;
    cin >> n;
    vector<int> heights(n);
    int m = 0;

    for (int i = 0; i < n; i++) {
        cin >> heights[i];
        m = max(m, heights[i]);
    }

    int ans = 0;
    for (int height = 1; height <= m; height++) {
        bool block_started = false;
        for (int i = 0; i < n; i++) {
            if (heights[i] < height) {
                block_started = true;
            }
            if (heights[i] == height && block_started && height > 0) {
                ans++;
                block_started = false;
            }
        }
        if (block_started) {
            ans++;
        }
    }
    cout << ans << endl;
    return 0;
}

File: solutions/other/indigo-code-2023/bblock.py
n = int(input())
heights = list(map(int, input().split()))

ans = 0

while any(heights):
    block_started = False
    for i, x in enumerate(heights):
        if x:
            block_started = True
        if x == 0 and block_started:
            ans += 1
            block_started = False
    if block_started:
        ans += 1
    for i in range(n):
        heights[i] = max(0, heights[i] - 1)

print(ans)
